data/net/minecraft/world/level/block/MultifaceBlock.mapping

```diff
@@ -1,12 +1,37 @@
 CLASS net/minecraft/world/level/block/MultifaceBlock
+	METHOD canSpreadInto (Lnet/minecraft/world/level/block/state/BlockState;)Z
+		ARG 1 state
+	METHOD getFaceProperty (Lnet/minecraft/core/Direction;)Lnet/minecraft/world/level/block/state/properties/BooleanProperty;
+		ARG 0 direction
 	METHOD getStateForPlacement (Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/world/level/BlockGetter;Lnet/minecraft/core/BlockPos;Lnet/minecraft/core/Direction;)Lnet/minecraft/world/level/block/state/BlockState;
 		ARG 1 currentState
 		ARG 2 level
 		ARG 3 pos
 		ARG 4 lookingDirection
+	METHOD hasAnyFace (Lnet/minecraft/world/level/block/state/BlockState;)Z
+		ARG 0 state
+	METHOD hasAnyVacantFace (Lnet/minecraft/world/level/block/state/BlockState;)Z
+		ARG 0 state
 	METHOD isFaceSupported (Lnet/minecraft/core/Direction;)Z
 		ARG 1 face
 	METHOD lambda$getStateForPlacement$1 (Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/world/level/Level;Lnet/minecraft/core/BlockPos;Lnet/minecraft/core/Direction;)Lnet/minecraft/world/level/block/state/BlockState;
 		ARG 4 lookingDirection
 	METHOD lambda$static$0 (Ljava/util/EnumMap;)V
 		ARG 0 map
+	METHOD mapDirections (Lnet/minecraft/world/level/block/state/BlockState;Ljava/util/function/Function;)Lnet/minecraft/world/level/block/state/BlockState;
+		ARG 1 state
+	METHOD spreadFromFaceTowardDirection (Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/world/level/LevelAccessor;Lnet/minecraft/core/BlockPos;Lnet/minecraft/core/Direction;Lnet/minecraft/core/Direction;Z)Z
+		ARG 1 state
+		ARG 2 levelAccessor
```

Comment:
```suggestion
		ARG 2 level
```

Reply:
Can you explain your reasoning on why this is better? I'm open to changing it, I just want to know why so I can learn for the future.

Reply:
Most of the usages of `Level` interfaces don't really have a specific reason for being typed as one of the interfaces as opposed to a different one - or at least, ones that we can clearly see. In most cases, these all refer to the same object (or, one of a few, i.e. the `level` in world gen is generally not the same as the `level` in server code vs. a `level` in client code.

In the absence of needing this parameter to be a specific interface, and since `level` is such a common occurrence throughout the code, these interfaces should generally be shortened just to `level`.

As an example where this *isn't* the case, look at `ChunkAccess.levelHeightAccessor`. This is explicitly typed very vaguely, as chunks need height information, but no guarantee is made that this is a normal `level` for generic use - and we can see that i.e. in `ProtoChunk`, this may contain a reference to the `ServerLevel`, which is a very big pitfall if you treat that as the currently generating level during world gen, because you'll create a circular chunk loading deadlock.